spikeinterface/core/base.py

```diff
@@ -526,10 +560,23 @@ def save_to_folder(self, name=None, folder=None, dump_ext='json', verbose=True,
         self.copy_metadata(cached)
 
         # dump
-        cached.dump(folder / f'cached.{dump_ext}')
+        cached.dump(folder / f'cached.{dump_ext}', relative_to=folder)
 
         return cached
-    
+
+
+def _make_paths_relative(d, relative):
+    dcopy = deepcopy(d)
+    if "kwargs" in dcopy.keys():
+        relative_kwargs = _make_paths_relative(dcopy["kwargs"], relative)
+        dcopy["kwargs"] = relative_kwargs
+        return dcopy
+    else:
+        for k in d.keys():
+            # in SI, all input paths have the "path" keyword
+            if "path" in k:
+                d[k] = str(Path(d[k]).relative_to(relative))
```

Comment:
OK I see. The error is here.
path can be str/Path or list of str/Path. You have to take this in account.
Please make a test with a multi segment object.

Reply:
Can you point me to an existing test with multi segment objects?

Reply:
https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/tests/testing_tools.py#L7